solr/core/src/java/org/apache/solr/update/UpdateLog.java

```diff
@@ -2002,13 +2009,15 @@ public void doReplay(TransactionLog translog) {
           translog.writeCommit(cmd);
         }
 
-        try {
-          proc.finish();
-        } catch (IOException ex) {
-          recoveryInfo.errors.incrementAndGet();
-          loglog.error("Replay exception: finish()", ex);
-        } finally {
-          IOUtils.closeQuietly(proc);
+        for (UpdateRequestProcessor proc : procPool) {
+          try {
+            proc.finish();
+          } catch (IOException ex) {
+            recoveryInfo.errors.incrementAndGet();
+            loglog.error("Replay exception: finish()", ex);
+          } finally {
+            IOUtils.closeQuietly(proc);
+          }
```

Comment:
Added commit here to have just the one for-loop rather than separate ones for `finish` and `close` -- if one `proc.finish` throws an exception then for the others `finish` would still be called. If multiple `proc.finish` were to throw then the errors counter would be incremented multiple times (rather than just once) but that seems less bad than the `finish` not being called for all the `proc` objects. WDYT?

Reply:
Nice!